tests/unit/test_synapse_workload.py

```diff
@@ -253,6 +333,103 @@ def test_enable_federation_domain_whitelist_error(
         synapse.enable_federation_domain_whitelist(container_mock, harness.charm._charm_state)
 
 
+def test_enable_trusted_key_servers_no_action(harness: Harness):
+    """
+    arrange: set mock container with file.
+    act: call enable_trusted_key_servers without changing the configuration.
+    assert: configuration is not changed.
+    """
+    root = harness.get_filesystem_root(synapse.SYNAPSE_CONTAINER_NAME)
+    config_path = root / synapse.SYNAPSE_CONFIG_PATH[1:]
+    config_path.parent.mkdir(parents=True, exist_ok=True)
+    config_path.write_text(
+        """
+listeners:
+    - type: http
+      port: 8080
+      bind_addresses:
+        - "::"
+"""
+    )
+
+    container: ops.Container = harness.model.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+    config = {"server_name": "foo"}
+    synapse_config = SynapseConfig(**config)  # type: ignore[arg-type]
```

Comment:
Would `SynapseConfig(server_name="foo")` work here? That way the type ignore is not needed

Reply:
If I change this then I get
```
tests/unit/test_synapse_workload.py:357: error: Missing named argument "federation_domain_whitelist" for "SynapseConfig"  [call-arg]
tests/unit/test_synapse_workload.py:357: error: Missing named argument "ip_range_whitelist" for "SynapseConfig"  [call-arg]
tests/unit/test_synapse_workload.py:357: error: Missing named argument "public_baseurl" for "SynapseConfig"  [call-arg]
tests/unit/test_synapse_workload.py:357: error: Missing named argument "report_stats" for "SynapseConfig"  [call-arg]
tests/unit/test_synapse_workload.py:357: error: Missing named argument "smtp_host" for "SynapseConfig"  [call-arg]
tests/unit/test_synapse_workload.py:357: error: Missing named argument "smtp_notif_from" for "SynapseConfig"  [call-arg]
tests/unit/test_synapse_workload.py:357: error: Missing named argument "smtp_pass" for "SynapseConfig"  [call-arg]
tests/unit/test_synapse_workload.py:357: error: Missing named argument "smtp_port" for "SynapseConfig"  [call-arg]
tests/unit/test_synapse_workload.py:357: error: Missing named argument "smtp_user" for "SynapseConfig"  [call-arg]
tests/unit/test_synapse_workload.py:357: error: Missing named argument "trusted_key_servers" for "SynapseConfig"  [call-arg]
```

Reply:
Doesn't that mean there is a problem here that this arguments are missing? I'm not sure how that constructor call works without providing those arguments

Reply:
I'll add this to be checked in one of the 2 stories that I created to refactor.

Reply:
Added to ISD-1423